File: algorithms/Yandex/algorithms_training_4/warm_up/C++/B.cpp
#include <iostream>
using std::cin, std::cout;

short gcd(const short& a, const short& b) {
    if (b == 0) return a;
    return gcd(b, a % b);
}


int inputB() {
    short a, b, c, d;
    cin >> a >> b >> c >> d;
    short m = a * d + c * b, n = b * d, div = gcd(m, n);
    cout << m / div << ' ' << n / div;
    return 0;
}

File: algorithms/Yandex/algorithms_training_4/warm_up/Python/B.py
def gcd(a: int, b: int):
    return a if b == 0 else gcd(b, a % b)


a, b, c, d = map(int, input().split())
m = a * d + c * b
n = b * d

d = gcd(m, n)
print(int(m / d), int(n / d))
